sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -131,7 +133,12 @@ public JobService getJobService(BigQueryOptions options) {
 
   @Override
   public DatasetService getDatasetService(BigQueryOptions options) {
-    return new DatasetServiceImpl(options);
+    return new DatasetServiceImpl(options, null);
+  }
+
+  @Override
+  public DatasetService getDatasetService(BigQueryOptions options, Histogram histogram) {
```

Comment:
I don't think we should be passing in the histogram to the factory for building the service. Couldn't we just instantiate it inside the call to newBigQueryClient()

Reply:
We create a new service instance per bundle, so it would be hard to aggregate the result across multiple bundles if we don't pass a histogram instance from outside of the service constructor.

We could also instantiate a histogram object inside the service constructor and merge them later but it's more time consuming (need an iteration over the bucket array).

Reply:
I see, could you rename the variable from "histogram" to something describing what the histogram is for, like "requestLatencies", "requestLatenciesHistogram"